ffc/billing/process_billing.py

```diff
@@ -211,42 +238,81 @@ async def process(self) -> AuthorizationProcessResult | None:
                     self.logger.info(f"No active agreement in the period {self.month}/{self.year}")
-                    result.errors.append(
+                    result_info.message = (
                         f"No active agreement for authorization {self.authorization_id}"
                     )
-                    return result
+
+                    result_info.result = ProcessResult.JOURNAL_SKIPPED
+                    return result_info
 
                 journal_external_id = f"{self.year:04d}{self.month:02d}"
                 filepath = self.build_filepath()
-
-                journal = await self.maybe_call(self.evaluate_journal_status, journal_external_id)
-                if journal and journal["status"] == VALIDATED:
-                    await self.maybe_call(self.mpt_client.submit_journal, journal["id"])
-                    return result
-
+                try:
+                    journal = await self.maybe_call(
+                        self.evaluate_journal_status, journal_external_id
+                    )
+                    if not journal:
+                        result_info.result = ProcessResult.ERROR
+                        result_info.message = (
+                            f"No journal found for external ID: {journal_external_id}"
+                        )
+                        return result_info
+                    status = journal["status"]
+                    if status == VALIDATED:
+                        await self.maybe_call(self.mpt_client.submit_journal, journal["id"])
+                        result_info.journal_id = journal["id"]
+                        result_info.result = ProcessResult.JOURNAL_GENERATED
+
+                    elif status != DRAFT:
+                        result_info.result = ProcessResult.JOURNAL_SKIPPED
+                        result_info.journal_id = journal["id"]
+                        return result_info
+
+                except JournalStatusError as error:
+                    self.logger.warning(error)
+                    result_info.message = str(error)
+                    result_info.result = ProcessResult.JOURNAL_SKIPPED
+                    result_info.journal_id = error.journal_id
+                    return result_info
                 self.logger.info(
                     f"generating charges file {filepath} "
                     f"currency {self.authorization['currency']}"
                 )
-                has_charges = await self.write_charges_file(filepath=filepath)
-                if has_charges:
-                    await self.maybe_call(
-                        self.complete_journal_process,
-                        filepath,
-                        journal,
-                        journal_external_id,
-                    )
-                    await self.maybe_call(aiofiles.os.unlink, filepath)
+                try:
+                    has_charges = await self.write_charges_file(filepath=filepath)
+                    if has_charges:
```

Comment:
We should return some Result in case for a given authorization the file is empty. Right now we are returning `None`. We should notify that for a given authorization, no charge file has been generated.